libraries/AP_Camera/AP_Camera_Backend.cpp

```diff
@@ -16,6 +16,17 @@ AP_Camera_Backend::AP_Camera_Backend(AP_Camera &frontend, AP_Camera_Params &para
 // update - should be called at 50hz
 void AP_Camera_Backend::update()
 {
+    // Check CAMx_OPTIONS and start/stop recording based on arm/disarm
+    if (_params.options.get() == CAMOPTIONS::REC_ARM_DISARM) {
+        if (hal.util->get_soft_armed() && !last_is_armed) {
```

Comment:
What you've got here is totally fine so just FYI, if it was me I would have made it a little more concise.  I probably would have written something like this:

```
if (hal.util->get_soft_armed() != last_is_armed) {
    last_is_armed = hal.util->get_soft_armed();
    record_video(last_is_armed);
}
```